src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -824,10 +830,36 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
         subscription_id,
         resource_group_name,
         {},
-        workspace_resource_id
+        workspace_resource_id,
+        appgw_name,
+        appgw_subnet_prefix,
+        appgw_id,
+        appgw_subnet_id,
+        appgw_shared,
+        appgw_watch_namespace
     )
     if 'omsagent' in addon_profiles:
         _ensure_container_insights_for_monitoring(cmd, addon_profiles['omsagent'])
+    if 'IngressApplicationGateway' in addon_profiles:
+        if "applicationGatewayId" in addon_profiles["IngressApplicationGateway"].config:
+            appgw_id = addon_profiles["IngressApplicationGateway"].config["applicationGatewayId"]
+            from msrestazure.tools import parse_resource_id, resource_id
+            appgw_id_dict = parse_resource_id(appgw_id)
+            appgw_group_id = resource_id(
+                subscription=appgw_id_dict["subscription"],
+                resource_group=appgw_id_dict["resource_group"])
+            if not _add_role_assignment(cmd.cli_ctx, 'Contributor',
+                                        service_principal_profile.client_id, scope=appgw_group_id):
+                logger.warning('Could not create a role assignment for IngressApplicationGateway addon. '
+                               'Are you an Owner on this subscription?')
+        if "subnetId" in addon_profiles["IngressApplicationGateway"].config:
+            subnet_id = addon_profiles["IngressApplicationGateway"].config["subnetId"]
+            from msrestazure.tools import parse_resource_id, resource_id
+            if not _add_role_assignment(cmd.cli_ctx, 'Contributor',
+                                        service_principal_profile.client_id, scope=subnet_id):
+                logger.warning('Could not create a role assignment for IngressApplicationGateway addon. '
```

Comment:
The error message is the same as above for appgw_group_id role assignment. Do you want to make it different?